src/mono/mono/component/CMakeLists.txt

```diff
@@ -57,14 +58,15 @@ list(APPEND components
   ${MONO_DIAGNOSTICS_TRACING_COMPONENT_NAME}
 )
 
-include(${MONO_EVENTPIPE_SHIM_SOURCE_PATH}/CMakeLists.txt)
+include(${MONO_EVENTPIPE_SHIM_SOURCE_PATH}/eventpipe.cmake)
 
 include_directories(
   ${MONO_EVENTPIPE_SHIM_SOURCE_PATH}
   ${MONO_EVENTPIPE_GEN_INCLUDE_PATH}
 )
 
 set(${MONO_DIAGNOSTICS_TRACING_COMPONENT_NAME}-sources
+  ${container_sources}
```

Comment:
I would agree. Ideally we would have a static library.

Reply:
>  a static library

I don't remember why I like object libraries so much... I think it was easier to think about symbol visibility... since an object library is just an extra pile of .o files on the linker's command line, I think `-fvisibility=hidden` on the consumer .so or executable will just do the right thing.  I think for static libs it's [more complicated](https://stackoverflow.com/questions/2222162/how-to-apply-fvisibility-option-to-symbols-in-static-libraries) (but maybe we already deal with it in our builds)